New variable $component-active-bg-light

stfnschrmr opened the pull request:
Adds separate variable `$component-active-bg-light` to avoid multiple use of `lighten($component-active-bg, 35%)` or `rgba($component-active-bg, .25)`. Also ensures that using native css custom properties as a value for `$component-active-bg` don't lead to an error when transpilling.

Commit:
Adds separate variable $component-active-bg-light to avoid multiple use of lighten($component-active-bg, 35%) or rgba($component-active-bg, .25). Also ensures that using native css custom properties as a value for $component-active-bg don't lead to an error when transpilling.

wolfy1339 reviewed (changes requested):
Missing a few spots here to change the variable names

stfnschrmr committed:
Update _variables.scss

All positions where `$component-active-bg` was altered by `lighten()` or `rgba()` get replaced with `$component-active-bg-light`.

MartijnCuppens commented:
Hi @createdwithlove, 
Nice to see you again. 

Caching some functioncalls in a variable is a good idea, but you can't just change the alpha version of the `$input-focus-border-color`, this will cause the border to disappear when the inputfield is focused.

This PR
![image](https://user-images.githubusercontent.com/11559216/47797454-a4e88e00-dd26-11e8-9d35-02048091da2b.png)

Bootstrap:
![image](https://user-images.githubusercontent.com/11559216/47797512-c3e72000-dd26-11e8-826f-0e878b1ea1d9.png)

It's a bit hard to see in the screenshots, but you can compare your version with the bootstrap docs:
- https://getbootstrap.com/docs/4.1/components/forms/#overview
- https://deploy-preview-27553--twbs-bootstrap4.netlify.com/docs/4.1/components/forms/#form-controls

createdwithlove commented:
Hi @MartijnCuppens.
You're right, that's a little detail that should not be lost. The calculation of the alpha value then could be done in the `form-control-focus()` mixin? But I don't know if that's worth it.

MartijnCuppens commented:
I don't think so, at the moment you can control the `border-color` and the `box-shadow` of the focus state and these colors can be changed independently. Calculating this in the mixin would make this impossible.

I'm going to close this because it's not worth adding another variable for this case. But thanks for the PR, if you spot something else that can be improved, let us know!
